python/ray/includes/serialization.pxi

```diff
@@ -347,7 +347,7 @@ cdef class Pickle5Writer:
     @cython.boundscheck(False)
     @cython.wraparound(False)
     cdef void write_to(self, const uint8_t[:] inband, uint8_t[:] data,
-                       int memcopy_threads) nogil:
```

Comment:
Why change this?

Reply:
Something went wrong with GIL, and let me update this comment after I make some more testing.

Reply:
Added back `nogil` to all `write_to` functions.

However, for function [L567](https://github.com/ray-project/ray/pull/35110/files#diff-6eb0b173395e77eb7364c2f68f0e4ab2abdd1875e73aeb420159d1149721af13R567) we have to wrap the code block with `with gil` since pyarrow functions are used.